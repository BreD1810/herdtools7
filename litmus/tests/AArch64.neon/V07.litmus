AArch64 V07
(* Tests LD3 single structure *)
{
x=65534;
0:X0=x; 0:X1=a; 0:X2=b; 0:X3=c; 0:X4=d; 0:X5=e;
}
 P0                             ;
 LD3 {V0.B,V1.B,V2.B}[1],[X0]   ;
 LD2 {V3.B,V4.B}[0],[X0]        ;
 ST1 {V0.B}[1],[X1]             ;
 ST1 {V1.B}[1],[X2]             ;
 ST1 {V2.B}[1],[X3]             ;
 ST1 {V3.B}[0],[X4]             ;
 ST1 {V4.B}[0],[X5]             ;
forall(a=254 /\ b=255 /\ c=0 /\ d=254 /\ e=255)
