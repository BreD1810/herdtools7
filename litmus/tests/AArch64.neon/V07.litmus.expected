Test V07 Required
Histogram (1 states)
1000000:>a=254; b=255; c=0; d=254; e=255;
Ok

Witnesses
Positive: 1000000, Negative: 0
Condition forall (a=254 /\ b=255 /\ c=0 /\ d=254 /\ e=255) is validated
Hash=7829a6f53dd26f46813c678bbbe6c87c
Observation V07 Always 1000000 0
